### PR TITLE
Log browser.OpenURL errors, try some hardcoded fallback browsers on Windows

### LINKED ISSUE #3261: github.com/pkg/browser.OpenURL can fail, BrowserOpenURL does not currently log errors, + fallbacks?
### Description

BrowserOpenURL calls github.com/pkg/browser.OpenURL.

github.com/pkg/browser.OpenURL can fail, but BrowserOpenURL ignores the error here.

Additionally, the current github.com/pkg/browser.OpenURL implementation for Windows is quite simple: https://github.com/pkg/browser/blob/master/browser_windows.go

```go
package browser

import "golang.org/x/sys/windows"

func openBrowser(url string) error {
	return windows.ShellExecute(0, nil, windows.StringToUTF16Ptr(url), nil, nil, windows.SW_SHOWNORMAL)
}
```

(I believe this is the same as entering a URL in the Windows Win+R "Run" dialog)

Unfortunately, this functionality breaks if the user's default browser is misconfigured. I'm not sure how users enter this state unintentionally, but I have steps to reproduce it below.

Ideas:

- Breaking change: BrowserOpenURL returns `error`, callers can handle this error as needed, or
- Non-breaking change: BrowserOpenURL logs the error
- Fallbacks: A more resilient implementation of BrowserOpenURL could be used. As an example, systems without a working default browser could fall back to trying a selection of hardcoded browser paths.

### To Reproduce

- Use Windows
- Set your default browser to Firefox (Settings -> Apps -> Default Apps -> Web Browser)
- Rename your firefox.exe to something.else
- In a Wails app, run `runtime.BrowserOpenURL("https://wails.io")`

### Expected behaviour

https://wails.io should open in the browser, or an error should be logged. 

### Screenshots

_No response_

### Attempted Fixes

https://github.com/wailsapp/wails/pull/3262 (probably an opinionated patch, but solved our issue)

### System Details

```shell
Wails Doctor          
                                

                                                                                                                                                                                              
# Wails
Version         | v2.8.0
Package Manager | pacman

# System
┌───────────────────────────────────────────────────────────────────────────────────────┐
| OS           | Manjaro Linux                                                          |
| Version      | Unknown                                                                |
| ID           | manjaro                                                                |
| Go Version   | go1.21.6                                                               |
| Platform     | linux                                                                  |
| Architecture | amd64                                                                  |
| CPU          | Intel(R) Core(TM) i7-8700T CPU @ 2.40GHz                               |
| GPU          | CoffeeLake-S GT2 [UHD Graphics 630] (Intel Corporation) - Driver: i915 |
| Memory       | 32GB                                                                   |
└───────────────────────────────────────────────────────────────────────────────────────┘

# Dependencies
┌─────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version     |
| *docker    | docker       | Installed | 1:24.0.7-1  |
| gcc        | gcc          | Installed | 13.2.1-3    |
| libgtk-3   | gtk3         | Installed | 1:3.24.39-2 |
| libwebkit  | webkit2gtk   | Installed | 2.42.4-1    |
| npm        | npm          | Installed | 10.3.0-1    |
| pkg-config | pkgconf      | Installed | 2.1.0-2     |
└────────────── * - Optional Dependency ──────────────┘

# Diagnosis
 SUCCESS  Your system is ready for Wails development!

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```


### Additional context

See also https://discord.com/channels/1042734330029547630/1207762177277034546

### PR DESCRIPTION
**Pull request by AlbinoDrought**
# Description

In all platforms, logs errors returned by `browser.OpenURL`.

On Windows, tries the additional hardcoded browser paths:

- `\Program Files (x86)\Microsoft\Edge\Application\msedge.exe`
- `\Program Files\Google\Chrome\Application\chrome.exe`
- `\Program Files (x86)\Google\Chrome\Application\chrome.exe`
- `\Program Files\Mozilla Firefox\firefox.exe`

Fixes #3261 

See also https://discord.com/channels/1042734330029547630/1207762177277034546

## Type of change
  
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Windows
- [ ] macOS
- [ ] Linux

> I was able to update the software with the new version you created and got in without issue. Thank you very much! 
> - Customer with invalid default browser config
  
## Test Configuration

- Use Windows
- Set your default browser to Firefox (Settings -> Apps -> Default Apps -> Web Browser)
- Rename your firefox.exe to something.else

# Checklist:

- [ ] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling and implemented fallback mechanisms for opening URLs across different desktop platforms.

### CODE REVIEW

File: v2/internal/frontend/desktop/darwin/browser.go [hunk 1]
```diff
@@ -10,5 +10,7 @@ import (
 // BrowserOpenURL Use the default browser to open the url
 func (f *Frontend) BrowserOpenURL(url string) {
 	// Specific method implementation
-	_ = browser.OpenURL(url)
+	if err := browser.OpenURL(url); err != nil {
+		f.logger.Error(err.Error())
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/darwin/browser.go:14]:
```suggestion
		f.logger.Error("Unable to open default system browser")
```

File: v2/internal/frontend/desktop/linux/browser.go [hunk 2]
```diff
@@ -8,5 +8,7 @@ import "github.com/pkg/browser"
 // BrowserOpenURL Use the default browser to open the url
 func (f *Frontend) BrowserOpenURL(url string) {
 	// Specific method implementation
-	_ = browser.OpenURL(url)
+	if err := browser.OpenURL(url); err != nil {
+		f.logger.Error(err.Error())
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/linux/browser.go:12]:
```suggestion
		f.logger.Error("Unable to open default system browser")
```

File: v2/internal/frontend/desktop/windows/browser.go [hunk 3]
```diff
@@ -5,10 +5,32 @@ package windows
...
 func (f *Frontend) BrowserOpenURL(url string) {
 	// Specific method implementation
-	_ = browser.OpenURL(url)
+	err := browser.OpenURL(url)
+	if err == nil {
+		return
+	}
+	for _, fallback := range fallbackBrowserPaths {
+		if err2 := openBrowser(fallback, url); err2 == nil {
+			f.logger.Warning("default browser.OpenURL failed but fell back to %v, original err: %v", fallback, err.Error())
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/windows/browser.go:27]: I'm not sure we need these. I'm wondering what sort of action would be taken based on seeing them.

File: v2/internal/frontend/desktop/windows/browser.go [hunk 4]
```diff
@@ -5,10 +5,32 @@ package windows
...
+			return
+		}
+	}
+	f.logger.Error("default browser.OpenURL failed, no fallback browser success, original err: %v", err.Error())
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/windows/browser.go:31]:
```suggestion
	f.logger.Error("Unable to open default system browser")
```

**Review by leaanthony — CHANGES_REQUESTED** [review]
Thanks for doing this! I checked the browser module we use and I think the error returned by it is fairly useless so suggest we provide our own error messages.

**Comment by AlbinoDrought** [comment]
(Sorry, not used to Github reviews)

Replaced error messages and removed warn on fallback: https://github.com/wailsapp/wails/compare/69375ca82904e0c32489f98ff6db6eb05ae542e8..cdeac67d3dc53df24178b92666fbb6b4b090d3ce

File: v2/internal/frontend/desktop/windows/browser.go (hunk 3 above)
Comment (AlbinoDrought) [reply v2/internal/frontend/desktop/windows/browser.go]: I agree, removed in https://github.com/wailsapp/wails/compare/69375ca82904e0c32489f98ff6db6eb05ae542e8..cdeac67d3dc53df24178b92666fbb6b4b090d3ce`

**Comment by leaanthony** [comment]
Thanks 🙏 Please could you add an entry to the changelog located at `website/src/pages/changelog.mdx`?

**Comment by leaanthony** [comment]
@AlbinoDrought - ping 😄

**Review by leaanthony — APPROVED** [review]

**Pull request merged**